connector/github/github.go

```diff
@@ -575,7 +580,21 @@ func (c *githubConnector) userEmail(ctx context.Context, client *http.Client) (s
 			}
 
 			if email.Verified && email.Primary {
-				return email.Email, nil
+				primaryEmail = email
+			}
+
+			if c.preferredEmailDomain != "" {
+				components := strings.Split(email.Email, "@")
+				if email.Verified && components[1] == c.preferredEmailDomain {
+					return email.Email, nil
+				}
+
+				// fallback to primary email if there is no preferred-domain email
+				if idx == len(emails)-1 {
```

Comment:
I'd like to suggest splitting logic.

1. Iterate through all domains and find the primary and preferred domain emails. Just find.
2. Decide what should be returned: primary email, email with preferred domain, or an error.